lint/ptr-to-ref-type-param_checker.go

```diff
@@ -0,0 +1,57 @@
+package lint
+
+import (
+	"go/ast"
+	"go/types"
+)
+
+func ptrToRefTypeParamCheck(ctx *context) func(*ast.File) {
+	return wrapFuncDeclChecker(&ptrToRefTypeParamChecker{
+		baseFuncDeclChecker: baseFuncDeclChecker{ctx: ctx},
+	})
+}
+
+type ptrToRefTypeParamChecker struct {
+	baseFuncDeclChecker
+}
+
+func (c *ptrToRefTypeParamChecker) CheckFuncDecl(fn *ast.FuncDecl) {
+	if fn.Type.Params != nil {
+		c.checkParams(fn.Type.Params.List)
+	}
+	if fn.Type.Results != nil {
+		c.checkParams(fn.Type.Results.List)
+	}
+}
+
+func (c *ptrToRefTypeParamChecker) checkParams(params []*ast.Field) {
+	for _, param := range params {
+		ptr, ok := c.ctx.TypesInfo.TypeOf(param.Type).(*types.Pointer)
+		if !ok {
+			continue
+		}
+
+		if isRefType(ptr.Elem().Underlying()) {
+			if len(param.Names) == 0 {
+				c.ctx.Warn(param, "consider to make returning type of non-pointer type")
```

Comment:
For input params, this will print confusing message about returning type.
It's OK for the first version though, just making an observation.

Reply:
`consider to make type of non-pointer type` ?

Reply:
That will work, yes.

It would be even better if type would be printer in error message.
Type string can be created with `nodeString(c.ctx.FileSet, param.Type)`.